proxy.js

```diff
@@ -16,13 +16,22 @@ var port = settings.port;
 var app = connect();
 var server = http.createServer(app);
 
+var MS_TIMEOUT = process.env.MS_TIMEOUT || 7000;
+var MAX_REQUESTS = process.env.MAX_REQUESTS || 30;
+
 var count = 0;
 
 app.use(compression())
 .use(function (req, resp) {
   count += 1;
   console.log('info active_requests=' + count);
 
+  if (count > MAX_REQUESTS) {
+    resp.statusCode = 503;
+    resp.end('Server under high load, try again later');
+    return;
```

Comment:
Will we increment the count for rejected requests but fail to decrement it on account of the early return? If so, we're in danger of continually issuing 503s if we get enough requests once hitting the threshold.